posthog/temporal/data_imports/pipelines/mysql/mysql.py

```diff
@@ -134,24 +141,29 @@ def _get_arrow_schema_from_type_name(table_structure: list[TableStructureRow]) -
 
     for col in table_structure:
         name = col.column_name
-        mysql_type = col.data_type
+        mysql_data_type = col.data_type
+        mysql_col_type = col.column_type
+
+        # Note that deltalake doesn't support unsigned types, so we need to convert integer types to larger types
+        # For example an uint32 should support values up to 2^32, but deltalake will only support 2^31
+        # so in order to support unsigned types we need to convert to int64
+        is_unsigned = "unsigned" in mysql_col_type
 
         arrow_type: pa.DataType
 
         # Map MySQL type names to PyArrow types
-        match mysql_type:
+        match mysql_data_type:
             case "bigint":
-                arrow_type = pa.int64()
+                arrow_type = pa.uint64() if is_unsigned else pa.int64()
```

Comment:
It looks like in the above comment and the corresponding switch statements below that you need more storage for the unsigned column type... but they appear to be the same here.

Reply:
well in this case there is no larger integer type (there's no `int128`) so not sure we have much choice. I can add a comment above just to make this clear though